File: Backend/KitchenDiary.API/Models/Recipe.cs
namespace KitchenDiary.API.Models;

public class Recipe
{
    public int Id { get; set; }

    public string Title { get; set; } = string.Empty;

    public string? Summary { get; set; }

    public decimal  Rating { get; set; }

    public string? Notes { get; set; }

    public DateTime DateAdded { get; set; }

    public ICollection<Ingredient> Ingredients { get; set; } = [];

    public ICollection<RecipeStep> Steps { get; set; } = [];

    public ICollection<RecipeImage> Images { get; set; } = [];

    public ICollection<RecipeTag> RecipeTags { get; set; } = new List<RecipeTag>();
    public string RecipeStyle { get; set; } = "";

    public string UserId { get; set; } = string.Empty;

    public ApplicationUser User { get; set; } = null!;
}

File: Frontend/kitchen-diary-ui/src/app/models/recipe.ts
export interface Recipe {
  id: number;
  title: string;
  summary: string;
  rating: number;
  notes: string;
  dateAdded: string;
  ingredients: Ingredient[];
steps: RecipeStep[];
  images: RecipeImage[];
}
export interface Ingredient {

  id: number;

  name: string;

  quantity: string | null;

}

export interface RecipeStep {

  id: number;

  stepNumber: number;

  instruction: string;

}

export interface RecipeImage {

  id: number;
  imagePath: string;

}

export interface Tag {

  id: number;
  name: string;

}

export interface Recipe {

  id: number;

  title: string;

  summary: string;

  rating: number;

  notes: string;

  dateAdded: string;

  ingredients: Ingredient[];

  steps: RecipeStep[];

  images: RecipeImage[];

  tags: Tag[];
  
  recipeStyle: string;

}
export interface RecipeImage {

  id: number;

  imagePath: string;

  isCoverImage: boolean;

}